l10n_nl_cbs_export/models/cbs_export_file.py

```diff
@@ -0,0 +1,279 @@
+                    "during month %s in year %s",
+                    calendar.month_name[int(cbs_export_file.month)],
+                    cbs_export_file.year
+                )
+            else:
+                cbs_export_file.export_file()
+
+    @api.multi
+    def export_file(self):
+        self.ensure_one()
+
+        cbs_export_data = self._format_header()
+        cbs_export_data += self._format_lines()
+        cbs_export_data += self._format_footer()
+
+        self.message_post(
+            body=_("CBS Export is created for month %s in year %s") % (
+                calendar.month_name[int(self.month)],
+                self.year
+            ),
+            subtype='mt_comment'
+        )
+        self.cbs_export_invoice = base64.b64encode(cbs_export_data.encode())
+
+    @api.model
+    def _format_header(self):
+        company = self.company_id
+        month_period = datetime.now().replace(
+            month=int(self.month), year=int(self.year))
+
+        cbs_export_data = \
+            str('9801') + \
+            str(company.vat or '').replace(' ', '')[2:].ljust(12) + \
+            str(month_period.strftime("%Y%m").ljust(6)) + \
+            str(company.name or '').ljust(40) + \
+            str(" " * 6) + \
+            str(" " * 5) + \
+            str(datetime.now().strftime("%Y%m%d").ljust(8)) + \
+            str(datetime.now().strftime("%H%M%S").ljust(6)) + \
+            str(company.phone or '').replace(
+                '+31', '0').replace(' ', '')[0:15].ljust(15) + \
+            str(" " * 13) + '\r\n'
+
+        return cbs_export_data
+
+    @staticmethod
+    def _format_footer():
+        cbs_export_data = str('9899') + str(" " * 111) + '\r\n'
+        return cbs_export_data
+
+    @api.multi
+    def _format_lines(self):
+        self.ensure_one()
+
+        line_counter = 1
+        invoice_line_number = 0
+        cbs_export_data = ''
+        invoices = self.account_invoice_ids
+
+        for invoice_line in invoices.mapped('invoice_line_ids'):
+            sign_of_weight = '-'
+            sign_of_invoice_value = '-'
+            if invoice_line.invoice_id.amount_total_signed >= 0:
+                sign_of_invoice_value = '+'
+            if (invoice_line.quantity * invoice_line.product_id.weight) >= 0:
+                sign_of_weight = '+'
+            if invoice_line.price_subtotal < 0:
+                sign_of_invoice_value = '-'
+
+            value = \
+                str(datetime.strptime(
+                    invoice_line.invoice_id.date_invoice, '%Y-%m-%d'
+                ).strftime("%Y%m") or '').ljust(6) + \
+                str('7') + \
+                str(invoice_line.company_id.vat or ''
+                    ).replace(' ', '')[2:].ljust(12) + \
+                str(line_counter).zfill(5) + str(" " * 3) + \
+                str(invoice_line.invoice_id.partner_id.country_id.code or ''
+                    ).ljust(3) + \
+                str('3') + \
```

Comment:
Why use str() here? It's not really clear by reading what is happening here.

Reply:
@tarteo I agree, the code should be refactored.

As explained [here](https://github.com/OCA/l10n-netherlands/pull/84#issuecomment-306746876):

> ... the fixed values we use are for most common situations correct

and in the readme:

> Field Transport "Vervoerwijze" is set hardcoded on 3. This is the most common used value.

I would like to do refactor and complete the module according to the Roadmap, but in a separate PR. What do you think?

Reply:
In a seperate PR is OK for me.